tools/benchmark/src/mocha/memoryTestRunner.ts

```diff
@@ -108,14 +115,27 @@ export interface MemoryTestObjectProps extends MochaExclusiveOptions, Titled, Be
 	 * @remarks {@link MemoryTestObjectProps.maxBenchmarkDurationSeconds} takes precedence over this, since a
 	 * benchmark with a very high measurement variance might never get a low enough RME.
 	 */
-	maxRelativeMarginOfError?: number;
+	readonly maxRelativeMarginOfError?: number;
 
 	/**
 	 * Percentage of samples (0.1 - 1) to use for calculating the statistics.
 	 * Defaults to 0.95.
 	 * Use a lower number to drop the highest/lowest measurements.
 	 */
-	samplePercentageToUse?: number;
+	readonly samplePercentageToUse?: number;
+
+	/**
+	 * The baseline memory usage to compare against for the test, which is used to determine if the test regressed.
+	 * If not specified, the test will not be compared against a baseline and will only be run to measure the memory usage.
+	 * @remarks Should be specified in bytes.
+	 */
+	readonly baselineMemoryUsage?: number;
+
+	/**
+	 * The allowed deviation from the `baselineMemoryUsage`, measured in bytes.
+	 * Has no effect if `baselineMemoryUsage` is not specified.
```

Comment:
```suggestion
	 * The allowed deviation from the `baselineMemoryUsage`, measured in bytes.
	 * @remarks
	 * Has no effect if `baselineMemoryUsage` is not specified.
	 * If `ENABLE_MEM_REGRESSION=1` in the environment, a test whose memory usage falls outside `baselineMemoryUsage +- allowedDeviationBytes` will be marked as failed.
	 * Otherwise, a warning is printed to the console.
```